pageserver/src/tenant/manifest.rs

```diff
@@ -0,0 +1,207 @@
+use std::io::{Read, Write};
+
+use crate::virtual_file::VirtualFile;
+use anyhow::Result;
+use bytes::{Buf, BufMut, Bytes, BytesMut};
+use crc32c::crc32c;
+use serde::{Deserialize, Serialize};
+use tracing::log::warn;
+use utils::lsn::Lsn;
+
+use super::storage_layer::PersistentLayerDesc;
+
+pub struct Manifest {
+    file: VirtualFile,
+}
```

Comment:
Would we need versioning, for easier backwards compatibility? What are the scenarios we'd want to be flexible to, and would it be easy to be backwards compatible?